### DISCUSSION TITLE
@PathVariable, @RequestParam 검증 결과를 가지고 있는 객체

### DISCUSSION THREAD

**Discussion by jehunyoo** [team-waff1e/Backend#11] [Q&A]
`BindingResult`는 타겟(`@ModelAttribute`)에서 발생한 예외(결과)를 가지고 있는 객체인데,
그럼 controller에서 `@PathVariable`, `@RequestParam`을 `@NotNull` 등의 어노테이션으로 검증한 결과는 어디서 찾을 수 있나요?

**Comment by suakang17** [comment]
spring-boot-starter-validation은 Hibernate Validator를 통해서 기능이 제공된다고 합니다

그래서 [해당문서](https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/)를 찾아 본 결과 다음과 같았습니다. (1.3. Validating constraints)

"The validate() method returns a set of ConstraintViolation instances, which you can iterate over in order to see which validation errors occurred. The first three test methods show some expected constraint violations" 

따라서 `ConstraintViolation`의 인스턴스에 저장되는 것으로 보입니다!

**Comment by jehunyoo** [comment]
추가로 찾아본 내용입니다.

> If the @PathVariable parameter fails validation, then Spring will add the error to the request's BindingResult automatically, you don't need to do that yourself.
https://stackoverflow.com/questions/4862420/problem-validating-pathvariable-url-on-spring-3-mvc
